logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```
@@ -0,0 +1,228 @@
+  private static final long STOP_TIMEOUT_IN_MS = 60000;
+  private static final int BASE_SLEEP_TIME_IN_MS = 50;
+  private static final int MAX_SLEEP_TIME_IN_MS = 30000;
+  private static final int MAX_NUM_RETRY = 20;
+
+  private final String mBaseLogsDir;
+  private int mPort;
+  private ServerSocket mServerSocket;
+  private final int mMinNumberOfThreads;
+  private final int mMaxNumberOfThreads;
+  private ExecutorService mThreadPool;
+  private volatile boolean mStopped;
+
+  /**
+   * Construct an {@link AlluxioLogServerProcess} instance.
+   *
+   * @param baseLogsDir base directory to store the logs pushed from remote Alluxio servers
+   */
+  public AlluxioLogServerProcess(String baseLogsDir) {
+    mPort = Configuration.getInt(PropertyKey.LOG_SERVER_PORT);
+    mMinNumberOfThreads = Configuration.getInt(PropertyKey.MASTER_WORKER_THREADS_MIN) + 2;
+    mMaxNumberOfThreads = Configuration.getInt(PropertyKey.MASTER_WORKER_THREADS_MAX) + 2;
+    mBaseLogsDir = baseLogsDir;
+    mStopped = true;
+  }
+
+  @Override
+  public void start() throws Exception {
+    startServing();
+  }
+
+  @Override
+  public void stop() throws Exception {
+    stopServing();
+  }
+
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mStopped == false;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  /**
+   * Create and start logging server and client thread pool.
+   */
+  private void startServing() {
+    SynchronousQueue<Runnable> synchronousQueue =
+        new SynchronousQueue<>();
+    mThreadPool =
+        new ThreadPoolExecutor(mMinNumberOfThreads, mMaxNumberOfThreads,
+            STOP_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS, synchronousQueue);
+    try {
+      mServerSocket = new ServerSocket(mPort);
+    } catch (IOException e) {
+      LOG.error("Failed to bind to port {}.", mPort);
+      return;
```

this should throw an exception - returning implies that the server has been started successfully

---

Done.